### DISCUSSION TITLE
403 error when trying to insert into table

### DISCUSSION THREAD

**Discussion by Kroplewski-M** [supabase/supabase#9335] [Questions]
have got a vue/supabase project that all im trying to do is register a user and add their details to a table. When trying to register the auth works and adds the email to the auth. The user can also log I'n as that works but when registering I try and add the email and username to a table and it keeps giving me a:

`POST https://vumblwlvddvktzbqrfkm.supabase.co/rest/v1/profiles?columns=%22username%22%2C%22email%22 403`

the table is simple, it has email and username columns that are text and the id is linked to the auth id. This is my add to table code which gives the 403 error:
` const addtoTable = async () => {
        try{
          const {data,error} = await supabase.from('profiles').insert([{username: username.value, email: email.value}], {returning: 'minimal'});
          if(error)throw error;
          alert("account added to table");
        }catch(error){ alert(error);}
      };`

**Comment by steve-chavez** [comment]
What's the `error` message?

**Reply by Kroplewski-M** [reply]
`OST https://vumblwlvddvktzbqrfkm.supabase.co/rest/v1/profiles?columns=%22username%22%2C%22email%22 403`
thats all it says in the console and the alert show shows `[object][object]`

**Reply by steve-chavez** [reply]
Try `console.log(JSON.stringify(error))`

**Reply by Kroplewski-M** [reply]
`{"code":"42501","details":null,"hint":null,"message":"new row violates row-level security policy for table \"profiles\""}`
oh this is the error how would I fix it?
thanks

**Reply by Kroplewski-M** [reply]
i disabled the RLS and now have a new error of: 
`{"code":"23502","details":"Failing row contains (null, 2022-10-04 21:19:55.753684+00, mat, mat@gmail.com).","hint":null,"message":"null value in column \"id\" of relation \"profiles\" violates not-null constraint"}`

**Reply by Kroplewski-M** [reply]
thats because i do not set the id I think,  but should that auto set itself from the auth user id?

**Reply by GaryAustin1** [reply]
You are likely getting an RLS error because you don't have a signed in user when you make this call.  Then when you turn off RLS you are getting an error because (I'm guessing you use auth.uid() as default for your id column) and because you have no signed in user that is null.

**Reply by Kroplewski-M** [reply]
i changed the order, i logged in first and then tried adding to table but get the same error of: 
`{"code":"23502","details":"Failing row contains (null, 2022-10-04 21:51:42.173165+00, mat, mat@gmail.com).","hint":null,"message":"null value in column \"id\" of relation \"profiles\" violates not-null constraint"}`

**Reply by GaryAustin1** [reply]
How are you setting the row to the user id?
If you can see your request in a network tab, you should also be able to see if the authorization header token is much longer than the anon key token, if it is the same, then you are not really signed in when you make the call.

**Reply by Kroplewski-M** [reply]
i set it with this sql command when I made the table `  id uuid references auth.users not null,`. When I log in using `const { user, session, error } = await supabase.auth.signIn()` I print out the auth and it says I'm authenticated: 
`{ "id": "04574e34-68ea-44b5-a991-9d63485800a1", "aud": "authenticated", "role": "authenticated", "email": "mat@gmail.com", "email_confirmed_at": "2022-10-04T20:18:35.846851Z", "phone": "", "confirmed_at": "2022-10-04T20:18:35.846851Z", "last_sign_in_at": "2022-10-04T21:59:10.605800221Z", "app_metadata": { "provider": "email", "providers": [ "email" ] }, "user_metadata": {}, "identities": [ { "id": "04574e34-68ea-44b5-a991-9d63485800a1", "user_id": "04574e34-68ea-44b5-a991-9d63485800a1", "identity_data": { "sub": "04574e34-68ea-44b5-a991-9d63485800a1" }, "provider": "email", "last_sign_in_at": "2022-10-04T20:18:35.844451Z", "created_at": "2022-10-04T20:18:35.844495Z", "updated_at": "2022-10-04T20:18:35.844499Z" } ], "created_at": "2022-10-04T20:18:35.841431Z", "updated_at": "2022-10-04T21:59:10.609386Z" }` and I have access to my ID, after that I try and add to table but get the error I showed you

**Reply by GaryAustin1** [reply]
You have to set the column in the table with a default of auth.uid().  Just making a reference does not set it, that just makes sure what you put there links to the auth.users table.

**Comment by Kroplewski-M** [comment]
yep when i set it manually now it works and its added to the table, thanks! i just thought it would set automatically when I register an account.
